README.md

```diff
@@ -79,10 +85,70 @@ const {
   positionals
 } = parseArgs({ args, options });
 console.log(values, positionals);
-// Prints: [Object: null prototype] { foo: true, bar: 'b' } []ss
+// Prints: [Object: null prototype] { foo: true, bar: 'b' } []
+```
+
+Detailed parse information is available for adding custom behaviours by specifying `tokens: true` in the configuration. The returned tokens have properties describing:
+
+* all tokens
+  * `kind`: { string } One of 'option', 'positional', or 'option-terminator'.
+  * `index`: { number } Index of element in `args` containing token.
+* option tokens
+  * `name`: { string } Long name of option.
+  * `rawName`: { string } How option used in args, like `-f` of `--foo`.
+  * `value`: { string | undefined } Option value specified in args. Undefined for boolean options.
+  * `inlineValue`: { boolean | undefined } Whether option value specified inline, like `--foo=bar`.
+* positional tokens
+  * `value`: { string } Positional value (i.e. `args[index]`).
+* option-terminator token
+
+For example, assuming the following script for `tokens.js`, which uses
+automatic detection of options and no error checking:
+
+```mjs
+import { parseArgs } from 'node:util';
+console.log(parseArgs({ strict: false, tokens: true }));
 ```
 
-`util.parseArgs` is experimental and behavior may change. Join the
+```cjs
```

Comment:
The `mjs` and `cjs` labels aren't visible, so this ends up being a bit weird - you say "the following script" but then there's two following scripts. (Also, the `mjs` one would need to be named `.mjs` for it to work without a `package.json` or an explicit `--input-type=module`.)

The other examples in this file all use the `cjs` style, so probably best to stick with that.

Reply:
This is a bit sneaky and might not work, but I noticed in the rendered upstream documentation there is only one script visible with a toggle: https://nodejs.org/dist/latest-v18.x/docs/api/util.html#utilparseargsconfig

> (Also, the mjs one would need to be named .mjs for it to work without a package.json or an explicit --input-type=module.)

So I fudged the command-line call by saying the script was called `tokens.js`, and left sorting out the cjs/mjs to the reader...

Reply:
I could rework without fudging the file extension? (I was copying some wording I found elsewhere and trying to make it fit here.)

Reply:
are linguist tags always file extensions? i'd expect `js` or `jsx` to work best

Reply:
Linguist understands `mjs` and `cjs` (and also `js`, `jsx`, and quite a few others besides): https://github.com/github/linguist/blob/d7799da826e01acdb8f84694d33116dccaabe9c2/lib/linguist/languages.yml#L3091-L3116